src/components/SearchForm.js

```diff
@@ -1,90 +1,23 @@
-import React, { useState } from "react";
-import uniq from "lodash/fp/uniq";
+import React from "react";
 
-const useInput = initialState => {
-  const [value, setValue] = useState(initialState);
-  const handleChange = e => {
-    setValue(e.target.value);
-  };
-  return [value, setValue, handleChange];
-};
-
-const SUGGESTED_LABELS = [
-  "good first issue",
-  "help wanted",
-  "first-timers-only"
-];
-
-const LANGUAGES = ["Any", "Haskell", "JavaScript", "OCaml", "PureScript"];
-
-function App() {
-  const [keywords, handleKeywordsChange] = useInput("");
-
-  const [label, setLabelValue, handleLabelChange] = useInput("");
-  const [labels, setLabels] = useState([]);
-
-  const [language, handleLanguageChange] = useInput(LANGUAGES[2]);
+import KeywordsInput from "./KeywordsInput";
+import LabelInput from "./LabelInput";
+import LanguageInput from "./LanguageInput";
 
+const SearchForm = () => {
   const handleSubmit = e => {
     e.preventDefault();
-    console.log({ keywords, labels, language });
+    console.log(e);
   };
 
-  const handleSubmitLabel = e => {
-    e.preventDefault();
-    if (label !== "") {
-      setLabels(ls => uniq([...ls, label.toLowerCase()]));
-      setLabelValue("");
-    }
-  };
   return (
     <form onSubmit={handleSubmit}>
-      <label>
-        <span className="label-text">Keywords</span>
-        <input value={keywords} onChange={handleKeywordsChange} />
-      </label>
-      <label>
-        <span className="label-text">With label</span>
-        <input
-          autoComplete="off"
-          list="labels"
-          value={label}
-          onChange={handleLabelChange}
-          onKeyDown={e => e.keyCode === 13 && handleSubmitLabel(e)}
-        />
-        <datalist id="labels">
-          {SUGGESTED_LABELS.filter(l => !labels.includes(l)).map(label => (
-            <option key={label} value={label} />
-          ))}
-        </datalist>
-        <ul>
-          {labels.map(label => (
-            <li key={label}>
-              <button
-                onClick={_e => setLabels(ls => ls.filter(l => l !== label))}
-              >
-                <span role="img" aria-label="X">
-                  ❌
-                </span>
-              </button>
-              {label}
-            </li>
-          ))}
-        </ul>
-      </label>
-      <label>
-        <span className="label-text">In language</span>
-        <select value={language} onChange={handleLanguageChange}>
-          {LANGUAGES.map(language => (
-            <option key={language} value={language}>
-              {language}
-            </option>
-          ))}
-        </select>
-      </label>
+      <KeywordsInput />
+      <LabelInput />
+      <LanguageInput />
       <button type="submit">Search →</button>
     </form>
   );
-}
+};
 
-export default App;
+export default SearchForm;
```

Comment:
Nice! This whole file just got a lot cleaner. We lost our access to the input state on form submit, but that's OK since we can hook this all up to Redux so we're not passing state between components.

Reply:
👍  Great. Glad to hear I was thinking on the right path here. I was working with Redux in the last big class project I did and got used to breaking things into small components and not worrying about passing state from child to parents. So we'll get there